src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -105747,5 +105747,190 @@ class Program
             var comp = CreateCompilation(source);
             comp.VerifyDiagnostics();
         }
+
+        [Fact]
+        [WorkItem(36131, "https://github.com/dotnet/roslyn/issues/36131")]
+        public void RunNullableWalker_GlobalDirective()
+        {
+            var source = @"
+class C
+{
+    void M(object? o1)
+    {
+        object o2 = o1;
+    }
+}";
+
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (6,21): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         object o2 = o1;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "o1").WithLocation(6, 21));
+
+            Assert.True(comp.RunNullableWalker);
+        }
+
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("annotations")]
+        [InlineData("warnings")]
```

Comment:
I like this use of Theory, I have a feeling I should have done the same for some of the directive tests